File: 2299-strong-password-checker-ii/2299-strong-password-checker-ii.cpp
class Solution {
public:
    bool strongPasswordCheckerII(string password) {
        int low=0;
        int up=0;
        int digit=0;
        int special=0;
        
        if(password.length()<8){ return false; }
        
        for(int i=0; i<password.length(); i++)
        {
            if(isupper(password[i])){ up++; }
            
            else if(islower(password[i])) { low++;}
            
            else if(isdigit(password[i])) { digit++; }
            
            else { special++; }
                
            if(i<password.length()-1 && password[i]==password[i+1]) { return false; }
        }
        
        return (low>0 && up>0 && digit>0 && special>0);
       
        
    }
};

File: 2299-strong-password-checker-ii/2299-strong-password-checker-ii.java
class Solution {
    public boolean strongPasswordCheckerII(String password) {
        if (password.length()<8) { return false; }
        
        int digit=0;
        int lowcase=0;
        int uppercase=0;
        int specialChar=0;
        
        for(int i=0; i<password.length(); i++)
        {
            char c=password.charAt(i);
            if(Character.isDigit(c)){ digit++; }
            else if( Character.isLowerCase(c)) { lowcase++; }
            else if(Character.isUpperCase(c)) { uppercase++; }
            else if(!Character.isAlphabetic(c)){ specialChar++; }
            
            if (i<password.length()-1 && c==password.charAt(i+1)) { return false; }
        }
        
        return (digit>0 && lowcase>0 && specialChar>0 && uppercase>0);
    }
}